social_simulator.py

```diff
@@ -16,24 +16,47 @@ def take_coords(ev):
     global kropki
     kropki += [Kropka([ev.x, ev.y])]
     for e in kropki:
-        while e == e.friend or e.friend == None:
-            e.friend = choice(kropki) if len(kropki)>1 else None
-        while e == e.enemy or e.enemy == None:
-            e.enemy = choice(kropki) if len(kropki)>1 else None
+        e.friend = choice(kropki) if len(kropki)>1 else None
+        while e == e.friend:
+            e.friend = choice(kropki)
+        e.enemy = choice(kropki) if len(kropki)>2 else None
+        while (e == e.enemy or e.friend == e.enemy) and e.friend != None:
+            e.enemy = choice(kropki)
+
+            
+           # if e == e.friend.enemy
 
 def switch(ev):
     global state
     state = not state
 
 def debug(ev):
-    if len(kropki)>1:
-        for e in kropki:
-            print('<kropka nr '+str(kropki.index(e))+' przyjaciel: '+str(kropki.index(e.friend))+', wróg: '+str(kropki.index(e.enemy)))
-        print('\n')
+    for e in kropki:
+        print('<kropka nr '+str(kropki.index(e))+' przyjaciel:',end=' ')
+        if e.friend in kropki:
+            print(str(kropki.index(e.friend)),end=' ')
+        else:
+            print('None',end = ' ')
+        print(', wróg: ',end = ' ')
+        if e.enemy in kropki:
+            print(str(kropki.index(e.enemy)))
+        else:
+            print('None')
+    print('')
+
+def getcolor():
+    return ("#"+''.join([choice('0123456789ABCDEF') for j in range(6)]))

```

Comment:
:clap: :clap: :clap:
`getcolor = lambda: '#'+''.join([choice('0123456789ABCDEF') for j in range(6)])` is even better.